subprojects/testing-junit-platform/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestExecutionListener.java

```
@@ -124,70 +144,20 @@ private void reportTestFailure(TestIdentifier testIdentifier, Throwable failure)
-        }
-    }
-
-    private static String reflectivelyReadExpected(Throwable failure) {
-        return reflectivelyRead(failure, "getExpected");
-    }
-
-    private static String reflectivelyReadActual(Throwable failure) {
-        return reflectivelyRead(failure, "getActual");
-    }
-
-    private static String reflectivelyRead(Object target, String methodName) {
-        String toStringMethod = isAssertionFailedErrorOrSubclass(target.getClass()) ? "getStringRepresentation" : "toString";
-        try {
-            Object value = target.getClass().getMethod(methodName).invoke(target);
-            return value == null ? null : (String) value.getClass().getMethod(toStringMethod).invoke(value);
-        } catch (Exception ignore) {
-            return null;
-        }
-    }
-
-    @SuppressWarnings("unchecked")
-    private static List<Throwable> getFailureListFromMultipleFailuresError(Throwable f) {
-        try {
-            String className = f.getClass().getCanonicalName();
-            if (className.equals("org.opentest4j.MultipleFailuresError")) {
-                Method getFailures = f.getClass().getMethod("getFailures");
-                return (List<Throwable>) getFailures.invoke(f);
-            } else if (className.equals("org.assertj.core.error.MultipleAssertionsError")) {
-                Method getFailures = f.getClass().getMethod("getErrors");
-                return (List<Throwable>) getFailures.invoke(f);
-            } else {
-                return null;
+    public TestFailure createFailure(Throwable failure) {
+        for (FailureMapper mapper : MAPPERS) {
+            if (mapper.supports(failure.getClass())) {
+                try {
+                    return mapper.map(failure, this);
+                } catch (Exception ignored) {
+                    // TODO before merge: Issue to do debug logging?
+                    // ignore
+                    System.out.println(ignored.getMessage());
```

❌ I'm fine with simply ignoring the exception here if it was the case previously.

---

PTAL again, now this uses a logger.
I'm on a fence on even ignoring this exception, as with the new mappers, you should not have a case when a mapper fail.

---

This should never happen, so reporting an error here makes sense to me.